src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -117,6 +118,11 @@ public void setBuyerRole(Person person, Buyer buyer) {
         addressBook.setBuyerRole(person, buyer);
     }
 
+    @Override
+    public void setSellerRole(Person person, Seller seller) {
+        requireAllNonNull(person, seller);
+        person.setSeller(seller);
```

Comment:
Following the setBuyerRole flow, maybe you can consider adding a setSellerRole(person, buyer) in address book too? Since in future increments we will probably add a UniqueBuyerList and UniqueSellerList field in addition to the current UniquePersonList in the AddressBook. Then we can maintain and add to the list from there on.

Reply:
good point, I'm now thinking if we should also have a separate UniqueBuyerList and UniqueSellerList, to facilitate searching for these roles